apps/challenges/utils.py

```diff
@@ -493,7 +498,8 @@ def add_tags_to_challenge(yaml_file_data, challenge):
         tags_data = yaml_file_data["tags"]
         new_tags = set(tags_data)
         # Remove tags not present in the YAML file
-        challenge.list_tags = [tag for tag in challenge.list_tags if tag in new_tags]
```

Comment:
Why this change? seems unrelated to the issue

Reply:
It was an oversight at my end.Rectified the changes.